src/components/ListItem.jsx

```diff
@@ -1,5 +1,63 @@
 import './ListItem.css';
+import { useCallback, useEffect, useState } from 'react';
+import { updateItem } from '../api/firebase';
+import { getFutureDate } from '../utils';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ listToken, item, itemId }) {
+	const { name, dateLastPurchased, checked } = item;
+	const [isPurchased, setIsPurchased] = useState(checked);
+
+	/**
+	 * If 24 hours has passed or the item is unchecked,
+	 * we change the state of the item and send to the database
+	 */
+	const handleChange = () => {
+		if (is24HoursPassed() || !checked) {
+			setIsPurchased(!isPurchased);
+
+			try {
+				updateItem(listToken, itemId, !isPurchased);
+			} catch (error) {
+				console.log(error);
+			}
+		}
+	};
+
+	/**
+	 * The is24HoursPassed function returns true of dateLastPurchased has passed 24 hours
+	 * Wrapping is24HoursPassed inside useCallback because this function is a dependency of useEffect
+	 */
+	const is24HoursPassed = useCallback(() => {
+		if (dateLastPurchased !== null) {
+			const dateLastPurchased_formatted = new Date(dateLastPurchased.toDate());
+			const yesterday = getFutureDate(-1);
+			return dateLastPurchased_formatted < yesterday;
+		}
+		return true;
+	}, [dateLastPurchased]);
+
+	// uncheck the item checked field and update the database value after 24 hours
+	useEffect(() => {
+		if (is24HoursPassed() && checked) {
+			setIsPurchased(false);
+			updateItem(listToken, itemId, false);
+		}
+	}, [is24HoursPassed, itemId, listToken, checked]);
+
+	return (
+		<>
+			<li className="ListItem">
+				<label aria-label="Mark as purchased">
```

Comment:
Great accessibility add-on here!
If I could make just one tiny suggestion: add also the name of the item to aria-live. The name is read by the screen reader if the user is not guiding the reader via the tab key, but if they do use the tab key, the reader will only announce "mark as purchased checkbox. unchecked". If name is added, then it reads "mark as purchased. cake checkbox. unchecked"

Reply:
Thank you for the suggestion! Just added the item name to aria-label. ✨